### ISSUE TITLE
Implement support for variable sequence length with the new extended CuDNN RNN ops

### ISSUE THREAD

**Issue by visionscaper** [keras-team/keras#12950]
This is a feature request.

Already since Tensorflow 1.12.1, support has landed for [extended cuDNN-RNN (supporting variable sequences) ops](https://github.com/tensorflow/tensorflow/commit/bf46fa4c5225e853e602b0e18951884b08729163). Enabling this support in Keras, would enable masking with CuDNNLSTM and CuDNNGRU layers, which would be very useful.

[The current code in Keras master](https://github.com/keras-team/keras/blob/cd3f27b5d2ccd843a2c98ee196f81217542e36c8/keras/layers/cudnn_recurrent.py#L66) shows that this has not been enabled yet.

**Comment by visionscaper** [comment]
I was wondering if anything is happening in this area?

**Comment by AlcoholV** [comment]
I'm wondering too. I really need this feature.

**Comment by AlcoholV** [comment]
It seems that someone suggested a solution.
https://github.com/tensorflow/tensorflow/issues/23269#issuecomment-531026

**Comment by atodniAr** [comment]
Is there anything being done to this problem? I know Keras supports variable input dimension within batches, but in some cases seems we can only use the padding-masking approach.
Say I'm working on a hierarchy LSTM model with input shape of `(n_batches, time steps_1, timesteps_2, input_dim)`, in which both timesteps are not fixed, then we can set the batch size to 1, and pass `None` for `timesteps_1`, but in order to pass a 3d array to Keras, we will still have to do padding over timesteps_2 and then mask it.
Any update on this? 🤔
